### ISSUE TITLE
onEnd handler for Audio Sprite seems bugged.

### ISSUE THREAD

**Issue by SlawomirMazgaj** [adireddy/waud#11]
Hi Adi, I was playing recently with an Audio Sprite and I'm getting some weird behaviors after adding `onEnd` handler. I was able to reproduce it using your sample. To do so:

**1)** In `_audSprite = new WaudSound("assets/sprite.json", {webaudio: false});` update webaudio value to true, so we can play few sounds from a sound sprite at the same time.

**2)** Add onEnd handlers:
```
_audSprite.onEnd(function(snd:IWaudSound) {trace("Glass finished.");}, "glass");
_audSprite.onEnd(function(snd:IWaudSound) {trace("Bell finished.");}, "bell");
_audSprite.onEnd(function(snd:IWaudSound) {trace("Canopening finished.");}, "canopening");
```

**3)** Add another magic button with this function as a handler:
```
function playAllTheThings() {
	Timer.delay( function() {_audSprite.play("glass");}, 100);
	Timer.delay( function() {_audSprite.play("bell");}, 200);
	Timer.delay( function() {_audSprite.play("canopening");}, 300);
}
```

So after pressing magic button I expect to see all 3 traces in JS console. Unfortunately I see `Canopening finished` 3 times. Am I doing something wrong or it's a bug?

**Comment by adireddy** [comment]
Hey @SlawomirMazgaj it's definitely a bug.

Are you expecting `onEnd` to fire on loop sounds as well?

**Comment by adireddy** [comment]
It's fixed in the latest release **0.5.8** and `onEnd` will not fire for loop sounds.

**Comment by SlawomirMazgaj** [comment]
Wow, that was quick, thanks.
We don't really use `onEnd` for looped sounds so 👍

**Comment by SlawomirMazgaj** [comment]
Verified fixed in version 0.5.8
